src/core/border_router/routing_manager.cpp

```diff
@@ -737,6 +739,11 @@ void RoutingManager::SendRouterAdvertisement(RouterAdvTxMode aRaTxMode)
         Error        error;
         Ip6::Address destAddress;
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_STUB_ROUTER_FLAG_IN_EMITTED_RA_ENABLE
+        SuccessOrAssert(raMsg.AppendFlagsExtensionOption(/* aStubRouterFlag */ true));
```

Comment:
I'm thinking we should append this at the top.

RFC 5175 states:
```
   o  MUST occur prior to any additional options associated with any
      flags set in this option.
```
And the stub router may need to process things like Prefix Information Options with an understanding of the stub-router flag value.

Reply:
Sounds good to me. Moved it to be the first option in RA message.